test/test_overrides.py

```diff
@@ -0,0 +1,756 @@
+    torch_vars = vars(torch)
+    untested_funcs = []
+    for func_name in torch.__all__ + dir(torch._C._VariableFunctions):
+        # these are functions that are public but shouldn't be overriden by __torch_function__
+        if func_name in IGNORED_TORCH_FUNCTIONS:
+            continue
+        # ignore private functions
+        if func_name.startswith('_'):
+            continue
+        # ignore in-place operators
+        if func_name.endswith('_'):
+            continue
+        # only consider objects with lowercase names
+        if not func_name.islower():
+            continue
+        func = torch_vars[func_name]
+        if func not in TENSOR_LIKE_OVERRIDES:
+            untested_funcs.append("torch.{}".format(func.__name__))
+    msg = (
+        "The following functions are not tested for __torch_function__ "
+        "support, please either add an entry in "
+        "TENSOR_LIKE_TORCH_IMPLEMENTATIONS for this function or if a "
+        "__torch_function__ override does not make sense, add an entry to "
+        "IGNORED_TORCH_FUNCTIONS.\n\n{}"
+    )
+    assert len(untested_funcs) == 0, msg.format(pprint.pformat(untested_funcs))
+    return TENSOR_LIKE_TORCH_IMPLEMENTATIONS
+
+@six.add_metaclass(ImplementationMeta)
+class TensorLike(object):
+    """A class that emulate the full tensor API
+
+    This class is used to explicitly test that the full torch.tensor API
+    can be overriden with a class that defines __torch_function__.
+    """
+    TORCH_IMPLEMENTATIONS = generate_tensor_like_torch_implementations()
+    IMPLEMENTS_DECORATOR = implements_tensor_like
+
+    def __torch_function__(self, func, args=(), kwargs=None):
+        if(kwargs is None):
+            kwargs = {}
+
+        if func not in HANDLED_FUNCTIONS_TENSOR_LIKE:
+            return NotImplemented
+        # In this case _torch_function_ should override TensorLike objects
+        return HANDLED_FUNCTIONS_TENSOR_LIKE[func](*args, **kwargs)
+
+class TestApis(TestCase):
+    def run(self, result=None):
+        """ Stop after first error """
+        if not result.errors:
+            super(TestApis, self).run(result)
+
+def test_generator(func, override):
+    args = inspect.getfullargspec(override)
+    nargs = len(args.args)
+    if args.defaults is not None:
+        nargs -= len(args.defaults)
+    func_args = [TensorLike() for _ in range(nargs)]
+    if args.varargs is not None:
+        func_args += [TensorLike(), TensorLike()]
+
+    msg = "torch.{} is not currently overridable".format(func.__name__)
+    @unittest.skipIf(func in TENSOR_LIKE_SKIP_TESTS, msg)
+    def test(self):
+        self.assertEqual(func(*func_args), -1)
+
+    return test
+
+if __name__ == '__main__':
+    for func, override in TENSOR_LIKE_TORCH_IMPLEMENTATIONS:
+        test_method = test_generator(func, override)
+        name = 'test_{}'.format(func.__name__)
+        test_method.__name__ = name
+        setattr(TestApis, name, test_method)
+    unittest.main()
```

Comment:
Have you tested this with pytest, too?

Reply:
No, do the tests need to work with pytest as well? I'm not sure if there's a way to do what I'm doing here in both pytest and with UnitTest.

Reply:
Want to Slack me about it?

Reply:
PyTest can run UnitTest suites, I believe.

Reply:
Tests now run under pytest, thanks for the pointer @mrubbery to the main `torch` tests for how to do this in a way that will work with pytest.